kratos/solving_strategies/builder_and_solvers/residualbased_elimination_builder_and_solver_with_constraints.h

```diff
@@ -0,0 +1,1522 @@
+    {
+        KRATOS_TRY
+
+        // First we set up the system of equations without constraints
+        BaseType::SetUpSystem(rModelPart);
+
+        // Add the computation of the global ids of the solvable dofs
+        IndexType counter = 0;
+        for (auto& dof : BaseType::mDofSet) {
+            if (dof.EquationId() < BaseType::mEquationSystemSize) {
+                auto it = mDoFSlaveSet.find(dof);
+                if (it == mDoFSlaveSet.end()) {
+                    ++counter;
+                }
+            }
+        }
+
+        // The total system of equations to be solved
+        mDoFToSolveSystemSize = counter;
+
+        KRATOS_CATCH("ResidualBasedEliminationBuilderAndSolverWithConstraints::FormulateGlobalMasterSlaveRelations failed ..");
+    }
+
+    /**
+     * @brief This method reconstructs the slave solution after Solving.
+     * @param pScheme The pointer to the integration scheme
+     * @param rModelPart Reference to the ModelPart containing the problem.
+     * @param rA System matrix
+     * @param rDx Vector of results (variations on nodal variables)
+     * @param rb RHS vector (residual)
+     */
+    void ReconstructSlaveSolutionAfterSolve(
+        typename TSchemeType::Pointer pScheme,
+        ModelPart& rModelPart,
+        TSystemMatrixType& rA,
+        TSystemVectorType& rDx,
+        TSystemVectorType& rb
+        )
+    {
+        KRATOS_TRY
+
+        // We get the global T matrix
+        TSystemMatrixType& rTMatrix = *mpTMatrix;
+
+        // We reconstruct the complete vector of Unknowns
+        VectorType Dx_copy(rDx);
+        rDx.resize(BaseType::mEquationSystemSize);
+        TSparseSpace::Mult(rTMatrix, Dx_copy, rDx);
+
+        // We reconstruct the LHS
+        if (mReassembleLHS) {
+            // We compute the transposed matrix of the global relation matrix
+            TSystemMatrixType T_transpose_matrix(mDoFToSolveSystemSize, BaseType::mEquationSystemSize);
+            SparseMatrixMultiplicationUtility::TransposeMatrix<TSystemMatrixType, TSystemMatrixType>(T_transpose_matrix, rTMatrix, 1.0);
+
+            // The auxiliar matrix to store the intermediate matrix multiplication
+            TSystemMatrixType auxiliar_A_matrix(BaseType::mEquationSystemSize, mDoFToSolveSystemSize);
+            SparseMatrixMultiplicationUtility::MatrixMultiplication(rA, T_transpose_matrix, auxiliar_A_matrix);
+
+            // We resize the LHS
+            rA.resize(BaseType::mEquationSystemSize, BaseType::mEquationSystemSize, false);
+
+            // Final multiplication
+            SparseMatrixMultiplicationUtility::MatrixMultiplication(rTMatrix, auxiliar_A_matrix, rA);
+        } else {
+            // Simply resize and reconstruct matrix structure
+            rA.resize(BaseType::mEquationSystemSize, BaseType::mEquationSystemSize, false);
+            ConstructMatrixStructure(pScheme, rA, rModelPart);
+        }
+
+        // Reconstruct the RHS
+        VectorType rb_copy(rb);
```

Comment:
because many things (convergence criteria for example) expect the RHS in his original size (try it, if you deactivate this you will have a memory error)